addons/html_editor/static/src/core/clipboard_plugin.js

```diff
@@ -181,38 +195,512 @@ export class ClipboardPlugin extends Plugin {
+            this.shared.domInsert(text);
+            return true;
+        }
+    }
+    /**
+     * @param {DataTransfer} clipboardData
+     */
+    onPasteOdooEditorHtml(clipboardData) {
+        const odooEditorHtml = clipboardData.getData("text/odoo-editor");
+        if (odooEditorHtml) {
+            const fragment = parseHTML(this.document, odooEditorHtml);
+            this.shared.sanitize(fragment, { IN_PLACE: true });
+            if (fragment.hasChildNodes()) {
+                this.shared.domInsert(fragment);
+            }
+        }
+    }
+    /**
+     * @param {EditorSelection} selection
+     * @param {DataTransfer} clipboardData
+     */
+    onPasteHtml(selection, clipboardData) {
+        const files = getImageFiles(clipboardData);
+        const clipboardHtml = clipboardData.getData("text/html");
+        if (files.length || clipboardHtml) {
+            const clipboardElem = this.prepareClipboardData(clipboardHtml);
+            // @phoenix @todo: should it be handled in table plugin?
+            // When copy pasting a table from the outside, a picture of the
+            // table can be included in the clipboard as an image file. In that
+            // particular case the html table is given a higher priority than
+            // the clipboard picture.
+            if (files.length && !clipboardElem.querySelector("table")) {
+                this.addImagesFiles(files).then((html) => {
+                    this.shared.domInsert(html);
+                });
+            } else {
+                if (closestElement(selection.anchorNode, "a")) {
+                    this.shared.domInsert(clipboardElem.textContent);
+                } else {
+                    this.shared.domInsert(clipboardElem);
+                }
+            }
+            return true;
+        }
     }
+    /**
+     * @param {EditorSelection} selection
+     * @param {DataTransfer} clipboardData
+     */
+    onPasteText(selection, clipboardData) {
+        const text = clipboardData.getData("text/plain");
+        let splitAroundUrl = [text];
+        // Avoid transforming dynamic placeholder pattern to url.
+        if (!text.match(/\${.*}/gi)) {
+            splitAroundUrl = text.split(URL_REGEX);
+            // Remove 'http(s)://' capturing group from the result (indexes
+            // 2, 5, 8, ...).
+            splitAroundUrl = splitAroundUrl.filter((_, index) => (index + 1) % 3);
+        }
+        if (splitAroundUrl.length === 3 && !splitAroundUrl[0] && !splitAroundUrl[2]) {
+            this.onPasteTextUrl(text, selection);
+        } else {
+            this.onPasteTextMultiUrl(splitAroundUrl, selection);
+        }
+    }
+    /**
+     * @param {string} text
+     * @param {EditorSelection} selection
+     */
+    onPasteTextUrl(text, selection) {
+        const selectionIsInsideALink = !!closestElement(selection.anchorNode, "a");
+        // Pasted content is a single URL.
```

Comment:
This comment should be next to `splitAroundUrl.length === 3 && !splitAroundUrl[0] && !splitAroundUrl[2]`.